app/controllers/form_controller.rb

```diff
@@ -213,6 +227,11 @@ def successful_redirect_path(pages_to_check)
       return send("#{@log.class.name.underscore}_#{previous_interruption_screen_page_id}_path", @log, { referrer: previous_interruption_screen_referrer, original_log_id: original_duplicate_log_id_from_query }.compact)
     end
 
+    if params[@log.model_name.param_key]["check_errors"]
+      @page = form.get_page(params[@log.model_name.param_key]["page"])
+      return send("#{@log.class.name.underscore}_#{params[@log.model_name.param_key]['original_page_id']}_path", @log, { check_errors: true, related_question_ids: params[@log.model_name.param_key]["related_question_ids"].split(" ") }.compact)
```

Comment:
If you can think of any way to make this a bit easier to read I'd love it, although I have failed to think of anything!

Reply:
Extracted some variables here as well, maybe a bit better: https://github.com/communitiesuk/submit-social-housing-lettings-and-sales-data/pull/2483